### PR TITLE
Migrate to changesets

### PR DESCRIPTION
**Pull request by AaronMoat**
Reasonably blindly copied https://github.com/seek-oss/logger/pull/89. Will need the token added (at a minimum)

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -32,12 +33,15 @@ jobs:
...
         env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GITHUB_TOKEN: ${{ secrets.SEEK_OSS_CI_GITHUB_TOKEN }}
           NPM_TOKEN: ${{ secrets.SEEK_OSS_CI_NPM_TOKEN }}
 
       - name: Publish Storybook
         run: yarn storybook:build && yarn storybook:deploy -- --ci --host-token-env-variable=GITHUB_TOKEN
         env:
-          GITHUB_TOKEN: ${{ github.actor }}:${{ secrets.GITHUB_TOKEN }}
+          GITHUB_TOKEN: ${{ github.actor }}:${{ secrets.SEEK_OSS_CI_GITHUB_TOKEN }}
```
Comment (72636c) [inline .github/workflows/release.yml:47]: I'm guessing we may need to remove `github.actor` if we want to make this change, as the org token belongs to https://github.com/orgs/seek-oss/teams/ci.

That said, can we check if we can stick to the repo-scoped GITHUB_TOKEN for the Storybook step? (I can't quite remember at this time why it's insufficient for Changesets though!)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -134,96 +134,69 @@ yarn test
...
 
-Consider whether you are making a visible change to the public Scoobie interface,
-which includes:
+We use [Changesets] to manage package releases.
+You'll see a 🦋 bot gliding around pull requests.
 
-- Top-level exports from [src/index.ts](/src/index.ts)
-- Top-level exports from [webpack/index.js](/webpack/index.js)
-- Bundled configuration files [types.d.ts](/types.d.ts) and [typography.ts](/typography.ts)
-- [npm dependencies](/package.json)
```
Comment (72636c) [inline CONTRIBUTING.md:145]: Should we keep this explanation of the "public interface", or do you reckon it's fairly obvious?

**Review by 72636c — APPROVED** [review]
🙇

File: .github/workflows/release.yml (hunk 1 above)
Comment (AaronMoat) [reply .github/workflows/release.yml:47]: I suppose if it already works it should be fine?

**Pull request merged**